src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentValidatorUTest.java

```diff
@@ -0,0 +1,148 @@
+        Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+        Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
+    }
+
+    @Test
+    public void validateCanvasAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+
+        assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+        Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
+    }
+
+    @Test
+    public void validateAttendanceAssignmentEmptyOptionalIdValidationError() throws IOException, AttendanceAssignmentException {
+        assignmentOptional = Optional.empty();
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+
+        assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+    }
+
+
+    @Test
+    public void validateAttendanceAssignmentNullCanvasIDError() throws IOException, AttendanceAssignmentException {
+        attendanceAssignment.setCanvasAssignmentId(null);
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+
+        assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+    }
+
+
+    @Test
+    public void validateCanvasAssignmentDBMismatchValidationError() throws IOException, AttendanceAssignmentException {
+        assignmentConfigurationFromSetup.setAssignmentPoints(120.0);
+
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID + "")).thenReturn(assignmentOptional);
+
+        try {
+            assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+            Assert.fail("Expected AttendanceAssignmentException.");
+        } catch (AttendanceAssignmentException exception) {
+            Assert.assertEquals(AttendanceAssignmentException.Error.NOT_SAVED, exception.error);
+        }
+    }
+
+    @Test
+    public void validateCanvasAssignmentEmptyAssignmentValidationError() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)